source/ConditionsStore.h

```diff
@@ -0,0 +1,152 @@
+/* ConditionsStore.h
+Copyright (c) 2020 by Peter van der Meer
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#ifndef CONDITIONS_STORE_H_
+#define CONDITIONS_STORE_H_
+
+#include <functional>
+#include <initializer_list>
+#include <map>
+#include <string>
+
+
+
+// Class that contains storage for conditions. Those conditions can be
+// set directly in internal storage of this class (primary conditions)
+// and those conditions can also be provided from other locations in the
+// code (derived conditions).
+//
+// The derived conditions are typically provided "on demand" from outside
+// this storage class. Some of those derived conditions could be read-only
+// and in a number of cases the conditions might be converted from other
+// data types than int64_t (for example double, float or even complex
+// formulae).
+class ConditionsStore {
+public:
+	// Storage types for conditions;
+	// - VALUE means that the condition-value is stored directly in this store.
+	// - PREFIX_PROVIDER means that conditions starting with the given prefix
+	//         are all handled by the given provider.
+	// - EXACT_PROVIDER means that the condition with the exact name is handled
+	//         by the given provider.
+	enum StorageType {VALUE, PREFIX_PROVIDER, EXACT_PROVIDER};
+
+	// Structure that describes the interface for DerivedProviders. When
+	// registering a new provider, then the (lambda)functions for accessing
+	// the conditions in the derived provider are communicated using this
+	// structure.
+	struct DerivedProvider {
+		std::function<int64_t(const std::string&)> getFun;
+		std::function<bool(const std::string&)> hasFun;
+		std::function<bool(const std::string&, int64_t)> setFun;
+		std::function<bool(const std::string&)> eraseFun;
+	};
```

Comment:
What would be a condition where `hasFun` and `eraseFun` would do something different than the default implementation? Especially `eraseFun`. Can't really delete a auto generated condition right?

Reply:
I expect nearly all auto-generated conditions to be read-only, but suppose that we make mortgages a read-write auto-generated condition, then erasing the condition that refers to a specific mortgage could remove that specific mortgage. (Not a very good example, because I expect the mortgages to be provided read-only if we make them available through conditions.)

Reply:
The [tributes](https://github.com/endless-sky/endless-sky/issues/5448) might be a better example.